kolena/_experimental/dataset/_evaluation.py

```diff
@@ -0,0 +1,225 @@
+from kolena._api.v2.model import LoadInferencesRequest
+from kolena._api.v2.model import Path
+from kolena._api.v2.model import UploadInferencesRequest
+from kolena._experimental.dataset._dataset import _iter_dataset_raw
+from kolena._experimental.dataset._dataset import _to_deserialized_dataframe
+from kolena._experimental.dataset._dataset import _to_serialized_dataframe
+from kolena._experimental.dataset.common import COL_DATAPOINT
+from kolena._experimental.dataset.common import COL_EVAL_CONFIG
+from kolena._experimental.dataset.common import COL_INFERENCE
+from kolena._experimental.dataset.common import COL_METRICS
+from kolena._experimental.dataset.common import validate_batch_size
+from kolena._utils import krequests_v2 as krequests
+from kolena._utils import log
+from kolena._utils.batched_load import _BatchedLoader
+from kolena._utils.batched_load import init_upload
+from kolena._utils.batched_load import upload_data_frame
+from kolena._utils.consts import BatchSize
+from kolena._utils.state import API_V2
+from kolena.errors import IncorrectUsageError
+
+
+@runtime_checkable
+class EvaluationConfiguration(Protocol):
+    def _to_dict(self) -> Dict[str, Any]:
+        ...
+
+
+TYPE_EVALUATION_CONFIG = Union[EvaluationConfiguration, Dict]
+INFER_FUNC_TYPE = Callable[[pd.DataFrame], pd.DataFrame]
+EVAL_FUNC_TYPE = Callable[[pd.DataFrame, pd.DataFrame, Optional[TYPE_EVALUATION_CONFIG]], pd.DataFrame]
+
+
+def _fetch_dataset(dataset: str) -> pd.DataFrame:
+    df_data_batch = list(_iter_dataset_raw(dataset))
+    return pd.concat(df_data_batch) if df_data_batch else pd.DataFrame(columns=["id", "datapoint"])
+
+
+def _iter_inference_raw(dataset: str, model: str, batch_size: int) -> Iterator[pd.DataFrame]:
+    validate_batch_size(batch_size)
+    init_request = LoadInferencesRequest(dataset=dataset, model=model, batch_size=batch_size)
+    yield from _BatchedLoader.iter_data(
+        init_request=init_request,
+        endpoint_path=Path.LOAD_INFERENCES.value,
+        df_class=None,
+        endpoint_api_version=API_V2,
+    )
+
+
+def _iter_metrics_raw(dataset: str, model: str, batch_size: int) -> Iterator[pd.DataFrame]:
+    validate_batch_size(batch_size)
+    init_request = LoadMetricsRequest(dataset=dataset, model=model, batch_size=batch_size)
+    yield from _BatchedLoader.iter_data(
+        init_request=init_request,
+        endpoint_path=EvaluationPath.LOAD_METRICS.value,
+        df_class=None,
+        endpoint_api_version=API_V2,
+    )
+
+
+def _fetch_inferences(dataset: str, model: str) -> pd.DataFrame:
+    df_inference_batch = list(_iter_inference_raw(dataset, model, batch_size=BatchSize.LOAD_RECORDS))
+    return (
+        pd.concat(df_inference_batch)
+        if df_inference_batch
+        else pd.DataFrame(
+            columns=["datapoint_id", COL_DATAPOINT, "inference_id", COL_INFERENCE],
+        )
+    )
+
+
+def fetch_inferences(dataset: str, model: str) -> Tuple[pd.DataFrame, pd.DataFrame]:
```

Comment:
Nit: although still under `_experimental`, but would still be nice to include docstrings now (admittedly I missed this earlier too)?